### PR TITLE
Move RouteManager.

### PR DESCRIPTION
**Pull request by siarheishut**
Move RouteManager from BusManager to RequestProcessor in order to ease Serialization implementation in future.

### CODE REVIEW

File: src/bus_manager.cpp [hunk 1]
```diff
@@ -27,13 +27,10 @@ int ComputeUniqueCount(std::vector<std::string> stops) {
 
 namespace rm {
 BusManager::BusManager(std::shared_ptr<TransportCatalog> transport_catalog,
+                       std::shared_ptr<RouteManager> route_manager,
                        const RoutingSettings &routing_settings)
```
Comment (foxfrikses) [inline src/bus_manager.cpp:31]: delete

File: src/bus_manager.cpp [hunk 2]
```diff
@@ -27,13 +27,10 @@ int ComputeUniqueCount(std::vector<std::string> stops) {
...
-    : transport_catalog_(transport_catalog) {
-
-  route_manager_ =
-      std::make_unique<RouteManager>(transport_catalog_->Stops(),
-                                     transport_catalog_->Buses(),
-                                     routing_settings);
+    : transport_catalog_(transport_catalog),
```
Comment (foxfrikses) [inline src/bus_manager.cpp:32]: move

File: src/bus_manager.cpp (hunk 1 above)
Comment (siarheishut) [reply src/bus_manager.cpp:31]: Done.

**Review by foxfrikses — APPROVED** [review]
LGTM

**Pull request merged**